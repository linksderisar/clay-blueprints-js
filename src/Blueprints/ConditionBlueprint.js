import Blueprint from './Blueprint';

export default class ConditionBlueprint extends Blueprint {
    constructor(type) {
        super(type);
        this._condition = '';
    }

    get condition() {
        return this._condition;
    }

    set condition(value) {
        this._condition = value;
    }

    toObject() {
        return {
            [this.type]: this.condition
        };
    }
}
