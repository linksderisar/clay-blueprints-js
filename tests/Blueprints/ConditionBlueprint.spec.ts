import ConditionBlueprint from '../../src/Blueprints/ConditionBlueprint';

it('basic condition works',  () => {
    const condition = new ConditionBlueprint('if');
    condition.condition = 'a === b';
    expect(condition).toMatchSnapshot({
        id: expect.any(String)
    });
});

it('allows empty conditions', () => {
    const condition = new ConditionBlueprint('if');
    expect(condition).toMatchSnapshot({
        id: expect.any(String)
    });
});
