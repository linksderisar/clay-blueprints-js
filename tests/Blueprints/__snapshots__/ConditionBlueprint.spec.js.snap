// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`allows empty conditions 1`] = `
Object {
  "_condition": "",
  "_type": "if",
  "id": Any<String>,
}
`;

exports[`basic condition works 1`] = `
Object {
  "_condition": "a === b",
  "_type": "if",
  "id": Any<String>,
}
`;
